### DISCUSSION TITLE
Need a hard/soft limit (or another solution) to mindwipes due to massive assistant powerscaling

### DISCUSSION THREAD

**Discussion by Xeavas** [FakeFishGames/Barotrauma#14858] [Unstable feedback]
Assistant can power scale hard by repeatedly completing the apprentice skill tree and using mindwipe to reset their skills while keeping bonus skills unlocked by graduating from said skill tree.

Potential solutions could be:
1. A hard cap + soft cap.
- e.g 1st reset free, 2nd -1 lvl -5% max vitality. Unable to reset the 3rd time

2. A soft cap, skill loss and/or permanent debuff(s) ramping up very fast making it increasingly punishing to reset skills.
- e.g. 1st reset free, 2nd -3 lvl, 3rd -6 lvl 
- e.g. 1st reset free, 2nd -1 lvl & -10% max vitality, 3rd -3lvl & additional -10% max vitality

3. Mindwipe removes bonus skills too (with some exceptions like endocrine skills).

I am personally more in favor of point 1 & 2 as it gives assistants some power scaling while not letting them infinitely able to do so in practice (in my opinion current -1 lvl every reset after 1st is too low of a debuff to offset this), on the other hand I am fine if point 3 is the solution if its deemed that it makes assistants too powerful & the concerns about modded bonus skills are addressed. 

Others I've talk to also mentioned using bonus skills in their mods so removing them via point 3 might be detrimental unless a solution to prevent it is implemented.

v1.6.17.0 when this post is made.

**Comment by Crazyjr20** [comment]
As it stands assistants have the potentional to have 18 talent trees (Albeit past 5 rerolls its gambling for duplicates).
Which is uh... yeah...
It takes 8 talent points to start off by giving assistant access to 3 talent trees, from then on every additional talent point lets you unlock an additional one from each class, meaning 13 talent points for 7 talent trees, and at that point you can expend an additional talent point for trying to not get a duplicate again and again, which is probably worth the risk compared to spending 4 points to unlock the other assistant trees fully (They aren't exactly good anyway).

As for how to solve this.
As said above either limit the amount of mindwipes fully imo or some more radical approachs like:
Saving the states of "Problem talents" so they cant be looped when unlocked again, but rather they just set it to how it was before. Course this has its own issue of limiting people from choosing a new apprenticeship.
Tagging additional talents added by "Problem talents" to remove them via mindwipe.

All I ask is however this is solved, don't take the nuclear approach of wiping all additional talents too, one this'll be a kick in the teeth for endocrine boosters, but two it'll break a lot of the way mods work to store permanent stats to characters including post death.

**Reply by Crazyjr20** [reply]
In addition to this, I have tested and confirmed that captain can constantly re-pick "Family" talent to generate infinite talent points for the rest of the crew, for themselves it's a neutral trade in talent points though, they neither lose or gain any.

**Comment by Regalis11** [comment]
This was something we were aware of and that we intentionally left in. We felt that having this sort of a "loophole" fits the assistants and would be a fun little extra gimmick for the class. We had a feeling it could be too powerful, but personally I thought it would be so difficult to gather enough mindwipes and level up a character this much, that it would be unlikely to be much of an issue in practice. But I suppose, if there's a "loophole" like this in the game, some players will find ways to make use of it. :)

I'm still on the fence about this, I'm not entirely sure how much time and effort it takes to actually use this to collect all/most of the talents - is it more of a theoretical problem, a character becoming OP if they spend dozens or hundreds of hours grinding for experience and mindwipes, or something that's likely to cause balance issues? And how much of an issue is it really if a character can become OP if they put a lot of work into using this "loophole"?

**Reply by Xeavas** [reply]
I think the resource sink isn't as significant of a bottleneck due to available methods to source significant amounts dementonite (abyss mining, buying from research stations, mainly abyss mining at high difficulties) or outright buying a mindwipe from research stations. 

It also could be said about captain T6 skill in Skipper tree "Family" that grants +1 levels to everyone upon getting the skill, thus will nullify skill lost for the captain while giving others and assistants in particular a way to quickly restore/gain levels repeatedly.

Though I think the main point arguing against leaving mindwipes like this is that it de-incentivizes leveling past 8-9 and getting other skill trees of assistants (and hyper incentivizes apprentice), as losing levels at higher levels require more exp to earn back a level whilst recovering levels when lower leveled requires less exp due to how experience scaling on levels works, because you're essentially getting an entire tree per level if you don't apprentice into the same class to not risk the chance of re-rolling the same tree.

So for example, one would need to spend 4 mind wipes and only 3 levels to get 4 additional trees :
-pick medic, get a medic tree
-reset, pick security and get a tree from that
-reset -1 level, pick engineer, get tree
-reset -1 level, pick mechanic, get tree
-reset -1 level, pick captain, get tree

**Reply by Ivanzyfer** [reply]
I think this is too strong a loophole, because one character can replace all the others while making them less useful, since they have only one branch of talents, some of which they cannot even get, since at some stages a choice is made between mutually exclusive talents. Playing with such a bug is similar to using cheats, which reduces the enjoyment of the game. You don't have to use this bug, but it looks like an artificial limitation, which also doesn't bring pleasure to the game process.
It is also not explained in any way from the point of view of the game world. If this thing erases memory, then all talents should be forgotten, not just the main ones, it would be logical and fair in relation to other professions that have no advantages over assistants. As far as I know, the role of assistant is recommended for beginners, but in fact it is the most powerful profession for experienced players who have the opportunity to become everything at once, which also disrupts the balance of the game and the distribution of roles where everyone contributes their part to the team game.

**Reply by Crazyjr20** [reply]
I mean a pretty punishing to indecisive people option would be to make the apprenticeship talent trees unlock trigger on an additional talent too, which is what I plan to do for a mod I'm working on that reworks more of the assistant talents. This way you cant reset it at all.

**Reply by KomushKrya** [reply]
> I think this is too strong a loophole, because one character can replace all the others while making them less useful, since they have only one branch of talents, some of which they cannot even get, since at some stages a choice is made between mutually exclusive talents. Playing with such a bug is similar to using cheats, which reduces the enjoyment of the game. You don't have to use this bug, but it looks like an artificial limitation, which also doesn't bring pleasure to the game process. It is also not explained in any way from the point of view of the game world. If this thing erases memory, then all talents should be forgotten, not just the main ones, it would be logical and fair in relation to other professions that have no advantages over assistants. As far as I know, the role of assistant is recommended for beginners, but in fact it is the most powerful profession for experienced players who have the opportunity to become everything at once, which also disrupts the balance of the game and the distribution of roles where everyone contributes their part to the team game.

I totally agree with Ivan's comment. From being a tool needed to redistribute talent points, it has evolved into the most powerful item in the game for a specific role. In order to fix this, it's enough just to erase all the talents of a person. Let him use this item several times in the hope of getting the right branch, rather than getting 6-7 or even more branches of all professions in a short time. Even in a vanilla game with high difficulty.

**Reply by Crazyjr20** [reply]
> I totally agree with Ivan's comment. From being a tool needed to redistribute talent points, it has evolved into the most powerful item in the game for a specific role. In order to fix this, it's enough just to erase all the talents of a person. Let him use this item several times in the hope of getting the right branch, rather than getting 6-7 or even more branches of all professions in a short time. Even in a vanilla game with high difficulty.

The problem with getting rid of "All the talents of a person" is that one in vanilla you are screwing over endocrine booster, but two for mods we often use talents as its the only persistent character based stat that isn't cleared on death.
In an ideal world, the talents given by apprenticeship are uniquely tagged, then those are removed, not all additional talents.

**Reply by Ivanzyfer** [reply]
What does it mean that only additional talents are removed? When erasing the memory, all the talents of the character should be erased. This is logical both from the point of view of the game world and from the point of view of balance.

**Reply by Crazyjr20** [reply]
> What does it mean that only additional talents are removed? When erasing the memory, all the talents of the character should be erased. This is logical both from the point of view of the game world and from the point of view of balance.

This would still mean every mod that needs to keep a permanent talent would now have to remove mindwipe on top to avoid being bricked.

I don't think its ideal to mess up additional talents you probably spent quite some time acquiring just because you wanted to respec. This is specifically a mindwipe abuse issue to me.
And the whole idea of refunding talent points doesn't make sense in the game world if your memory has been wiped, unless said talents were related to being a peak physical strength.
Obviously it would be silly to make it only refund select talents though from a gameplay point of view and would make the whole point of the item void as we already have the create new character option.

All this is to just say either fix Mindwipe specifically (I.E, make it a permanent choice which I dont think is the ideal solution but I know can be achieved in current xml as I've just got it working) or add the ability to flag talents when added as a "dont refund on mindwipe". (Its worth mentioning the function apprenticeship uses for unlocking talents is unique to itself, so could be addressed within that.)

**Reply by Ivanzyfer** [reply]
When creating a new character, you need to upgrade it again to get talents. I'm not saying that you need to reset the character's level, but the additional talents gained from the assistant's talent should disappear. These are different things. When talents are reset, all talents must be reset so that they can be reallocated. Additionally received talents should be erased at the same time, that is, they can be obtained anew if you select the right talent again.

**Reply by Crazyjr20** [reply]
I agree that the additional talents from Mindwipe should be wiped, but additional talents in general no.

**Reply by Ivanzyfer** [reply]
Of course, the number of points to explore talents should remain consistent with the character's level.

**Reply by sebastianyarrik** [reply]
Assistants just can unlock captain skills, get free points and increase experience for low level persons . Five assistants can give +50% for every assistant and +250% experience for guy with minimal level. Assistants with politican talents tree can use medals for keeping talents in five level. Finally, balance absolutely broken.

**Reply by mathiascasaer** [reply]
with respect but there is simply no reason on a serious campaign to not run an army of assistants. we ran long campaigns regularly for 1300+ hours in the past. 

On our current (ironman) campaign our army of assistants is jugging down mindwipes like crazy and it renders every single other class as a subpar. you say its hard to get mindwipes? did you actually ever play a campaign in your own game recently? the recipe is trivial. + you can buy it everywhere.  the deterrant (xp loss) is negatable with a stacked xp bonus.  you say you are aware and intentionally left this in.  

' I'm not entirely sure how much time and effort it takes to actually use this to collect all/most of the talents - is it more of a theoretical problem, a character becoming OP if they spend dozens or hundreds of hours grinding for experience and mindwipes, or something that's likely to cause balance issues?' 

theorethical?  i tell you in good faith : we get to that state in **less than 8 hours play**, with indentured servitude and bots covering all the class essentials, my other players are all assistants,  this is trivial , easy, and, contagious: in short: it completely fubars the balance , its the worst power creep feature/bug ive seen in any game ever and its realy disappointing and simply no fun to play other classes on a mid- or endgame. you opened pandoras box. fun little gimmick... 

how to fix? idk... either provide a way for other classes to get their mutually exclusive talents , or to idk; get a skill boost to their native skill that can not be aquired as an assistant. or tone it down at the risk of community backlash..

i find it hard to understand how a new update was done without addressing this, given the ingenuity and the well thought out balance of baro in EA and the fact that this thread shows that this has been reported even at the time of unstable, and eloquently , as f.e. by Ivan. if you _still_ doubt it, you're always welcome to hop in on our current campaign, just ask Pike for our discord (ask for Haze)  and come check it out so you can get a feel of it.

**Reply by Regalis11** [reply]
@mathiascasaer (and others in this thread too!), thank you for the well-argued feedback. I think there's more than enough good arguments here to explain why this loophole is problematic (even if you're not a very experienced player who's really good at minmaxing and playing the game in an "optimal" way). I've brought this up with our designers again, I think we'll discuss ways to address this - the solution that seems the most simple to me is to just make mindwipes remove all the extra talents unlocked via the Apprentice tree, but I suppose we'll need to have a think about whether that'd have some negative side-effects or balance implications.

**Reply by Crazyjr20** [reply]
@Regalis11 As long as you don't blanket hit all Additional Talents _(This would hit Endocrine Booster and mods using talents as a through death variable in the crossfire, but by your wording it doesn't seem like you would hit all Additional Talents)_ it would be the best route in my opinion.

Being able to change your apprenticeship tree would discourage people from reloading saves as much when they get the wrong tree like prior to Mindwipe existing.
People will still be able to abuse it to get recipes but it would be extremely cumbersome to do that as you have the cost of mindwipe, stacking level loss and having to gamble again for the right talent tree you originally had, so I don't see that as a huge issue.


While my ideal solution would be the removal of talents added by the apprenticeship like you said, there is a couple other options that could be considered:

One being to single out the problem and hit graduation ceremony, making it not be able to be Mindwiped. Though I don't like this method as it will encourage reloading saves again and feels a bit cruel to be single out this one talent as a "You must live with your mistake" talent.

Another being to once again single out graduation ceremony, but change its functionality to not give talents, but rather change it so it gives different buffs based on apprenticeship. This one is more work having to fully rework it and for likely a much more boring talent concept though that would probably not be as easy a pill to swallow for assistant lovers.

All this being said I love the concept of Mindwipe, letting people change their mind without having to dive into the CharacterData xml to manually do it. It's just its highly abusable at current due to talents originally not being designed with being reset in mind, making problem ones like Apprenticeship and originally Family.

**Reply by sjTaylor** [reply]
Dunno about other people,

but I was planning on using mindwipe to let me "pick" my specialization for Graduation Ceremony.
(I recently swapped to Assistant in the game I'm playing with some friends)

It was actually disappointing in a way to find out it cumulatively stacks talents.
This loophole feels more like cheating to me than a feature.
Even if it is technically an intentional feature of the game.

Personally, my issue would be "fixed" by making Graduation Ceremony just not random.
Without looking into save editing, I'd need keep making new characters to get the tree I want without risking becoming better than the original class at their stuff.

I also think it would be good in general to allow Graduation Ceremony to be picked (instead of random) since it makes some classes just way way worse to pick as an apprenticeship IMO.


To Summarize: My ideal solution would be Graduation Ceremony letting the player select, and Mind Wipe getting rid of the talents it gives (so it doesn't accumulate talents indefinitely)

**Comment by EternalPainZXC** [comment]
I don't see a problem with the fact that the assistant, as the weakest class, has the right to grind.
Achieve great power through long and hard work.
If other players think that this is unfair, let them try playing in a multiplayer campaign by choosing the assistant.
The main gaming experience of playing for the assistant consists of constant deaths, which makes it impossible to gain experience, not to mention erasing the mind.
In other words, rising above level 3 is already a feat.
If after this the players think that the assistant is still unbalanced, let them try playing on any difficulty except easy.
I think that the ticket should be closed before it turns into a trash thread.
With great respect, mrktyr.

**Reply by Ivanzyfer** [reply]
At maximum difficulty, you can easily avoid dying on an assistant if you play as a team and reach any level.
The problem is precisely in the balance, since it makes no sense to take on other roles when there is an unbalanced assistant.

**Reply by MKMoose** [reply]
Please.

> I don't see a problem with the fact that the assistant, as the weakest class, has the right to grind.
> Achieve great power through long and hard work.

Try to understand the full extent of the issue at hand before saying it shouldn't be fixed.
The problem lies largely in that Assistant can get disproportionately strong _without even grinding_. In an optimal scenario an Assistant can gain 27-29 additional Talents at least about as fast as other jobs can finish their second tree. The power ceiling, which includes 84 Talents from other jobs and 29 of Assistant's own Talents, though it can take a long time to get to due to randomness, is literally about as high as the combined ceiling of all other jobs. It can be argued to be even higher, due to Assistant's ability to unlock pairs of Talents that are otherwise mutually exclusive.

> If other players think that this is unfair, let them try playing in a multiplayer campaign by choosing the assistant.

Assistant being weak is in no significant way related to this issue. If anything, this issue only exacerbates the problem with Assistant's Talents, because it makes one of the subtrees incomparably better than the other two.
Plus, playing the Assistant is literally how people found that this is a serious balance issue.

> The main gaming experience of playing for the assistant consists of constant deaths, which makes it impossible to gain experience, not to mention erasing the mind.
> In other words, rising above level 3 is already a feat.

Other jobs don't have anything in particular that consistently saves them from death. If a person's constantly dying as an Assistant, I really don't think that's _only_ the Assistant being weak at fault.

> If after this the players think that the assistant is still unbalanced, let them try playing on any difficulty except easy.

Difficulty changes very little in moment-to-moment situations until you get to Vitality multipliers and therefore doesn't drastically increase the frequency of deaths that would be prevented by Assistant getting buffed.
Do note as well that the majority of people talking about this issue haven't played on easy since their first campaign or ever and have had more than enough time to experience the unbalanced powerhouse that is Assistant on two dozen Mindwipes, whereas many relatively inexperienced people that are playing on easy either won't know about Mindwipe or will get their experience largely ruined in a certain capacity if they use it, since it can easily trivialize the game for them at a time when they are expected to be familiarizing themselves with more engaging game mechanics.

> I think that the ticket should be closed before it turns into a trash thread.

This discussion has been here for more than half a year. Think about that.

**Comment by Regalis11** [comment]
This has been addressed in our private repo now by making mindwipe remove the extra talents assistants have unlocked using the "Apprentice" tree. The change will be included in the next update.

Thank you for all the feedback and discussion! :)

**Reply by mathiascasaer** [reply]
ty , excellent, much appreciated.

**Reply by Regalis11** [reply]
The change has now been tested internally, coming up in the next unstable build.

**Reply by Gittanley** [reply]
another pure brimstone by fekalis

**Reply by abolog0** [reply]
I myself accidentally found this bug once and was pleasantly surprised, because it gave me many more opportunities for leveling up my character especially since I don’t like the clown branch, and the left branch seems too weak compared to the others. I would buff it somehow or change some of its skills. I can also suggest making the third assistant branch a choice between the Church of Hask and the Clown one in my opinion, it would be more interesting to choose between these two competing factions, since there aren’t really any skills related to the Church of Hask for other characters (the medic has some genes, but I don’t count that because it’s very superficial, while the clowns have a whole branch) (sorry my eng)
